tka/state.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tka
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"golang.org/x/crypto/argon2"
+)
+
+// ErrNoSuchKey is returned if the key referenced by a KeyID does not exist.
+var ErrNoSuchKey = errors.New("key not found")
+
+// State describes Tailnet Key Authority state at an instant in time.
+//
+// State is mutated by applying Authority Update Messages (AUMs), resulting
+// in a new State.
+type State struct {
+	LastAUMHash        []byte   `cbor:"1,keyasint"`
```

Comment:
I think these fields could use some commentary.

Are hashes and secrets always fixed lengths? I'm wondering if we should be using named array types for them `type AUMHash [32]byte`. That would mean using zeros instead of nil, but might be particularly nice with a `type Secret [64]byte` type.